pkg/ipam/crd.go

```diff
@@ -346,11 +349,7 @@ func (n *nodeStore) updateLocalNodeResource(node *ciliumv2.CiliumNode) {
 		var allocator *crdAllocator
 		var ipFamily Family
 		if ipAddr := net.ParseIP(ip); ipAddr != nil {
-			if ipAddr.To4() != nil {
-				ipFamily = IPv4
-			} else {
-				ipFamily = IPv6
-			}
+			ipFamily = DeriveFamily(ipAddr)
```

Comment:
This could be part of the first commit, no?

Reply:
The first and second commits are now squashed.